app/controllers/application_controller.rb

```diff
@@ -2,4 +2,18 @@ class ApplicationController < ActionController::Base
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
+
+  rescue_from Exception, with: :handle_exception
+
+  before_filter :add_allow_credentials_headers
+
+
+  def handle_exception(exception)
+    redirect_to :root
+  end
+
+  def add_allow_credentials_headers
```

Comment:
in my opinion method name not reflects inside actions.
I think method should be renamed to something like that: specified_credentials_in_headers